sdk/core/Azure.Core/samples/Response.md

```diff
@@ -86,3 +86,28 @@ catch (RequestFailedException e) when (e.Status == 404)
     Console.WriteLine("ErrorCode " + e.ErrorCode);
 }
 ```
+
+### Exception details
+
+Some services may provide additional details when a service call fails. Apart from the error `Code` and `Message` that are part of the service contract, you should only use this information for diagnostics
+since the values could change any time.
+
+```C# Snippet:RequestFailedExceptionDetails
+try
+{
+    KeyVaultSecret secret = client.SetSecret("DeletedSecret", "secret");
+}
+// handle exception with status code 404
+catch (RequestFailedException e) when (e.Status == 409)
+{
+    ResponseError error = e.GetRawResponse().Content.ToObjectFromJson<ResponseError>();
```

Comment:
I think this doesn't work today, if a service is following the Azure API guidance and using the outer `error` block.

Reply:
Very good point! I forgot about that and brought it up with @JoshLove-msft in a conversation about this topic previously. As such, I think we should also provide a model that just has a `public ResponseError Error` property, but even since when we briefly discussed this previously, I still haven't come up with a good name. Initially I was thinking "ErrorResponse" since it's a response with an `error`, but 1) we already have an `ErrorResponse<T>` so that's confusing, and 2) customers may not be entirely clear on which to use as the top-level model: `ErrorResponse` or `ResponseError`. Any suggestions?